auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/PublicDeltaLoginController.kt

```diff
@@ -1,18 +1,118 @@
 package uk.gov.communities.delta.auth.controllers
 
 import io.ktor.server.application.*
+import io.ktor.server.request.*
 import io.ktor.server.response.*
 import io.ktor.server.thymeleaf.*
+import org.slf4j.LoggerFactory
 import uk.gov.communities.delta.auth.config.DeltaConfig
+import uk.gov.communities.delta.auth.security.ADLdapLoginService
+import uk.gov.communities.delta.auth.security.LdapUser
 
-class PublicDeltaLoginController {
-    suspend fun getLoginPage(call: ApplicationCall) {
-        call.respond(
-            ThymeleafContent(
-                "delta-login", mapOf(
-                    "deltaUrl" to DeltaConfig.DELTA_WEBSITE_URL
-                )
+
+class PublicDeltaLoginController(private val ldapService: ADLdapLoginService) {
+    private val logger = LoggerFactory.getLogger(this.javaClass)
+
+    suspend fun loginGet(call: ApplicationCall) {
+        call.respondLoginPage()
+    }
+
+    private suspend fun ApplicationCall.respondLoginPage(
+        errorMessage: String = "", errorLink: String = "#", username: String = "", password: String = ""
+    ) = respond(
+        ThymeleafContent(
+            "delta-login",
+            mapOf(
+                "deltaUrl" to DeltaConfig.DELTA_WEBSITE_URL,
+                "errorMessage" to errorMessage,
+                "errorLink" to errorLink,
+                "username" to username,
+                "password" to password,
             )
         )
+    )
+
+    suspend fun loginPost(call: ApplicationCall) {
+        val formParameters = call.receiveParameters()
+        val formUsername = formParameters["username"]
+        val password = formParameters["password"]
+
+        if (formUsername.isNullOrEmpty()) return call.respondLoginPage(
+            errorMessage = "Username is required", errorLink = "#username"
```

Comment:
Respond with the login page directly since we don't have sessions yet. Ideally also we'd display both username and password required if they're both empty, but I've kept it simple for now.